pkg/homedir/homedir_linux.go

```diff
@@ -62,41 +63,56 @@ func stick(f string) error {
 	return os.Chmod(f, m)
 }
 
+func getHomeDir() (string, error) {
+	home := os.Getenv("HOME")
+	if home != "" {
+		return home, nil
+	}
+
+	currentUser, err := user.Current()
+	if err == nil {
+		return currentUser.HomeDir, nil
+	}
+
+	return "", errors.New("could not get either HOME or the passwd home directory")
+}
+
 // GetDataHome returns XDG_DATA_HOME.
-// GetDataHome returns $HOME/.local/share and nil error if XDG_DATA_HOME is not set.
+// GetDataHome returns $HOME/.local/share and nil error if XDG_DATA_HOME is not set and HOME is set.
+// GetDataHome returns user.Current().HomeDir/.local/share and nil error if XDG_DATA_HOME and HOME are not set.
 //
 // See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
 func GetDataHome() (string, error) {
 	if xdgDataHome := os.Getenv("XDG_DATA_HOME"); xdgDataHome != "" {
 		return xdgDataHome, nil
 	}
-	home := os.Getenv("HOME")
-	if home == "" {
-		return "", errors.New("could not get either XDG_DATA_HOME or HOME")
+	if home, err := getHomeDir(); err == nil {
```

Comment:
Perhaps this should just use `Get()`? (we can still detect if it's empty); https://github.com/moby/moby/blob/62296f9281d429b0ff1967f4c933e4b3b338cdd8/pkg/homedir/homedir_unix.go#L25-L33

Reply:
Thanks, this is a much better solution. We now use `Get()` instead of `os.Getenv("HOME").